collector/factory.go

```diff
@@ -69,7 +69,10 @@ func BuildProfilesReceiver(options ...option) xreceiver.CreateProfilesFunc {
 			ExecutableReporter:     controllerOption.executableReporter,
 			OnShutdown:             controllerOption.onShutdown,
 		}
-
+		err := controlerCfg.Validate()
```

Comment:
Could you add a test that creates an invalid factory?

Reply:
Unit test added